### PR TITLE
perf: change replay payload to reduce json size

### PR DESCRIPTION
**Pull request by anonrig**
Depends on https://github.com/getsentry/snuba/pull/5831

### CODE REVIEW

File: src/sentry/replays/lib/event_linking.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ class EventLinkKafkaMessage(TypedDict):
     replay_id: str
     project_id: int
     segment_id: None
-    payload: list[int]
+    payload: str
```
Comment (asottile-sentry) [inline src/sentry/replays/lib/event_linking.py:20]: I almost think this should be a union between the various payload dictionaries that are expected on the rust size -- smuggling through bytes as `list[int]` and double-json encoding both seem like hacks

**Review by asottile-sentry — APPROVED** [review]
needs [this](https://github.com/getsentry/snuba/pull/5831) first but looks good -- thanks for fixing this double-json!

**Comment by JoshFerge** [comment]
why are we doing this?

**Comment by anonrig** [comment]
> why are we doing this?

@JoshFerge Because we pass unoptimized data to Snuba. For example:

```
(Pdb) ret_value['payload']
[123, 34, 116, 121, 112, 101, 34, 58, 34, 101, 118, 101, 110, 116, 95, 108, 105, 110, 107, 34, 44, 34, 114, 101, 112, 108, 97, 121, 95, 105, 100, 34, 58, 34, 49, 55, 57, 51, 55, 102, 99, 51, 54, 55, 55, 101, 52, 48, 48, 97, 56, 57, 49, 53, 53, 51, 101, 52, 100, 55, 54, 53, 101, 50, 98, 97, 34, 44, 34, 101, 118, 101, 110, 116, 95, 104, 97, 115, 104, 34, 58, 34, 55, 48, 100, 51, 102, 57, 54, 54, 45, 57, 50, 55, 55, 45, 99, 97, 98, 54, 45, 102, 54, 102, 54, 45, 98, 52, 57, 55, 100, 53, 51, 49, 99, 52, 53, 49, 34, 44, 34, 116, 105, 109, 101, 115, 116, 97, 109, 112, 34, 58, 49, 55, 49, 52, 52, 49, 55, 48, 51, 55, 44, 34, 101, 114, 114, 111, 114, 95, 105, 100, 34, 58, 34, 101, 50, 98, 102, 99, 50, 97, 50, 102, 49, 101, 57, 52, 50, 100, 54, 56, 48, 101, 48, 98, 97, 98, 52, 99, 54, 97, 101, 52, 101, 49, 57, 34, 125]
(Pdb) bytes(ret_value['payload'])
b'{"type":"event_link","replay_id":"17937fc3677e400a891553e4d765e2ba","event_hash":"70d3f966-9277-cab6-f6f6-b497d531c451","timestamp":1714417037,"error_id":"e2bfc2a2f1e942d680e0bab4c6ae4e19"}'
```

**Review by JoshFerge — APPROVED** [review]

**Review by asottile-sentry — APPROVED** [review]
thanks for fixing the tests too!

**Comment by untitaker** [comment]
https://sentry.sentry.io/issues/5283168570/?project=300688&query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D&referrer=issue-stream&statsPeriod=90d&stream_index=1

**Comment by untitaker** [comment]
dependency PR in snuba didn't go out, reverting

**Pull request merged**